zio-http/src/main/scala/zio/http/Handler.scala

```diff
@@ -516,14 +535,23 @@ sealed trait Handler[-R, +Err, -In, +Out] { self =>
     Http.fromHandler(self)
 
   /**
-   * Converts a Handler into a websocket application
+   * Creates a new response from a socket app.
    */
-  final def toSocketApp(implicit
-    ev1: WebSocketChannelEvent <:< In,
-    ev2: Err <:< Throwable,
+  def toResponse(implicit
+    ev1: Err <:< Throwable,
+    ev2: WebSocketChannel <:< In,
     trace: Trace,
-  ): SocketApp[R] =
-    SocketApp(event => self.runZIO(event).mapError(ev2))
+  ): ZIO[R, Nothing, Response] =
+    ZIO.environment[R].flatMap { env =>
+      Response.fromSocketApp(self.asInstanceOf[SocketApp[R]].provideEnvironment(env))
+    }
+
+  def toRoute(implicit
```

Comment:
```suggestion
  def toHttpimplicit
```

Reply:
We already have a `toHttp` operator with a different signature that propagates the `In` type instead of one that takes a `Handler` with an `In` type of `WebSocketChannel` and converts it into an `Http` with an `In` type of `Any` so that it can be further composed. This is where we are feeling a little bit the deletion of `SocketApp` which had these more specialized operators.

Reply:
Maybe `toHttpFromSocketApp`?

Reply:
Or `ToHttpApp`.